### PR TITLE
PMM-3479 Error message for different versions.

### PR DESCRIPTION
**Pull request by BupycHuk**
Print error message if pmm-server and client has different versions.

### CODE REVIEW

File: pmm/pmm.go [hunk 1]
```diff
@@ -552,6 +553,55 @@ func (a *Admin) checkSSLCertificate() error {
 	return generateSSLCertificate(a.Config.ClientAddress, SSLCertFile, SSLKeyFile)
 }
 
+// ServerPing is a class for data from SERVER_URL/ping.
```
Comment (AlekSi) [inline pmm/pmm.go]: Please use `/managed/v1/version` for URL. That API will stay in 2.0.

File: pmm/pmm.go [hunk 2]
```diff
@@ -552,6 +553,55 @@ func (a *Admin) checkSSLCertificate() error {
...
+			cleanedErr := strings.Replace(err.Error(), a.serverURL, serverURL, -1)
+			return true, fmt.Errorf(`Unable to connect to PMM server by address: %s
+%s
+
+* Check if the configured address is correct.
+* If server is running on non-default port, ensure it was specified along with the address.
+* If server is enabled for SSL or self-signed SSL, enable the corresponding option.
+* You may also check the firewall settings.`, a.Config.ServerAddress, cleanedErr)
+		}
+		if Version > ping.Version {
```
Comment (AlekSi) [inline pmm/pmm.go]: Please use https://github.com/percona/pmm/blob/master/version/version_info.go. The same code will be used in 2.0.

File: pmm/managed/client.go [hunk 3]
```diff
@@ -172,3 +172,12 @@ func (c *Client) ScrapeConfigsDelete(ctx context.Context, jobName string) error
 func (c *Client) AnnotationCreate(ctx context.Context, req *APIAnnotationCreateRequest) error {
 	return c.do(ctx, "POST", "/v0/annotations", req, nil)
 }
+
+// VersionGet returns version of the managed API.
+func (c *Client) VersionGet(ctx context.Context) (*VersionResponse, error) {
+	res := new(VersionResponse)
+	if err := c.do(ctx, "GET", "/v1/version/", nil, res); err != nil {
```
Comment (AlekSi) [inline pmm/managed/client.go]: Remove trailing `/` – it is not a collection.

File: pmm/pmm.go [hunk 4]
```diff
@@ -552,6 +554,37 @@ func (a *Admin) checkSSLCertificate() error {
 	return generateSSLCertificate(a.Config.ClientAddress, SSLCertFile, SSLKeyFile)
 }
 
+// CheckVersion check server and client versions.
```
Comment (AlekSi) [inline pmm/pmm.go]: Please document return parameters.

**Comment by rnovikovP** [comment]
@AlekSi  pls review this

**Comment by AlekSi** [comment]
@BupycHuk PTAL

**Comment by BupycHuk** [comment]
@AlekSi looks good to me

**Comment by AlekSi** [comment]
https://github.com/Percona-Lab/pmm-submodules/pull/166

File: pmm/pmm.go [hunk 5]
```diff
@@ -552,6 +554,59 @@ func (a *Admin) checkSSLCertificate() error {
...
+	if err != nil {
+		return true, err
+	}
+
+	// Return fatal error if major versions do not match.
+	// Texts are slightly different, including anchors.
+	if serverVersion.Major < clientVersion.Major {
+		return true, fmt.Errorf(
+			"Error: You cannot run PMM Server %d.x with PMM Client %d.x.\n"+
+				"Please upgrade Server by following the instructions at "+
```
Comment (michaelcoburn) [inline pmm/pmm.go]:
```suggestion
				"Please upgrade PMM Server by following the instructions at "+
```

File: pmm/pmm.go [hunk 6]
```diff
@@ -552,6 +554,59 @@ func (a *Admin) checkSSLCertificate() error {
...
+				"https://www.percona.com/doc/percona-monitoring-and-management/deploy/index.html#deploy-pmm-updating",
+			serverVersion.Major, clientVersion.Major,
+		)
+	}
+	if serverVersion.Major > clientVersion.Major {
+		return true, fmt.Errorf(
+			"Error: You cannot run PMM Server %d.x with PMM Client %d.x.\n"+
+				"Please upgrade Client by following the instructions at "+
```
Comment (AlekSi) [inline pmm/pmm.go]:
```suggestion
				"Please upgrade PMM Client by following the instructions at "+
```

File: pmm/pmm.go [hunk 7]
```diff
@@ -552,6 +554,59 @@ func (a *Admin) checkSSLCertificate() error {
...
+				"https://www.percona.com/doc/percona-monitoring-and-management/deploy/index.html#updating",
+			serverVersion.Major, clientVersion.Major,
+		)
+	}
+
+	// Return warning if versions do not match.
+	if serverVersion.Less(&clientVersion) {
+		return false, fmt.Errorf(
+			"Warning: The recommended upgrade process is to upgrade PMM Server first, then clients.\n" +
```
Comment (AlekSi) [inline pmm/pmm.go]:
```suggestion
			"Warning: The recommended upgrade process is to upgrade PMM Server first, then PMM Clients.\n" +
```

File: pmm/pmm.go [hunk 8]
```diff
@@ -552,6 +554,59 @@ func (a *Admin) checkSSLCertificate() error {
...
+				"See Percona's instructions for upgrading at " +
+				"https://www.percona.com/doc/percona-monitoring-and-management/deploy/index.html#deploy-pmm-updating.",
+		)
+	}
+	if clientVersion.Less(&serverVersion) {
+		return false, fmt.Errorf(
+			"Warning: It is recommended to use the same version on both PMM Server and Client, otherwise some features will not work correctly.\n" +
+				"Please upgrade your Client by following the instructions from " +
```
Comment (AlekSi) [inline pmm/pmm.go]:
```suggestion
				"Please upgrade your PMM Client by following the instructions from " +
```

**Pull request merged**